plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.eclipse.dataflow.ui.preferences;
+
+import com.google.cloud.tools.eclipse.dataflow.core.preferences.DataflowPreferences;
+import com.google.cloud.tools.eclipse.dataflow.ui.page.MessageTarget;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RunOptionsDefaultsComponentTest {
+  
+  private @Mock DataflowPreferences preferences;
+  private Shell shell;
+  private MessageTarget messageTarget = Mockito.mock(MessageTarget.class);
+  private RunOptionsDefaultsComponent component;
+  private Composite composite;
+
+  @Before
+  public void setUp() {
+    shell = new Shell(Display.getCurrent());
+    composite = new Composite(shell, SWT.NONE);
+    component = new RunOptionsDefaultsComponent(composite, 3, messageTarget, preferences);
+  }
+
+  @After
+  public void tearDown() {
+    if (shell != null && !shell.isDisposed()) {
```

Comment:
`shell` is never null (always created in `setUp()`), and it is okay to call `dispose()` multiple times, so you don't need this if-stmt.